com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/shader/ShaderCompilePipeline.java

```diff
@@ -439,23 +468,30 @@ public byte[] crossCompile(ShaderDesc.ShaderType shaderType, ShaderDesc.Language
             FileUtil.deleteOnExit(fileCrossCompiled);
 
             generateWGSL(module.desc.resourcePath, module.spirvFile.getAbsolutePath(), fileCrossCompiled.getAbsolutePath());
-            return FileUtils.readFileToByteArray(fileCrossCompiled);
+
+            Shaderc.ShaderCompileResult result = new Shaderc.ShaderCompileResult();
+            result.data = FileUtils.readFileToByteArray(fileCrossCompiled);
+            return result;
         } else if (CanBeCrossCompiled(shaderLanguage)) {
             Shaderc.ShaderCompileResult result = generateCrossCompiledShader(shaderType, shaderLanguage, version);
 
             if (!result.lastError.isEmpty()) {
                 throw new CompileExceptionError("Cross-compilation of shader type: " + shaderType + ", to language: " + shaderLanguage + " failed, reason: " + result.lastError);
             }
 
-            byte[] bytes = result.data;
-
             // JG: spirv-cross renames samplers for GLSL based shaders, so we have to run a second pass to force renaming them back.
             //     There doesn't seem to be a simpler way to do this in spirv-cross from what I can understand.
             if (ShaderLanguageIsGLSL(shaderLanguage)) {
-                bytes = RemapTextureSamplers(module.spirvReflector.getTextures(), new String(bytes));
+                result.data = RemapTextureSamplers(module.spirvReflector.getTextures(), new String(result.data));
+            }
+
+            // If we are cross-compiling a compute shader to HLSL, we need to inject an extra resource into the reflection,
+            // because there is no gl_NumWorkGroups equivalent in HLSL
+            if (shaderType == ShaderDesc.ShaderType.SHADER_TYPE_COMPUTE && result.hLSLNumWorkGroupsId >= 0) {
+                injectHLSLNumWorkGroupsIdIntoReflection(module, result.hLSLNumWorkGroupsId);
             }
```

Comment:
Hm, one problem with this is if we need to store multilple hlsl files in the same shader file (e.g building for windows + xbox), they could potentially be stored at different bind points but only exist once in the reflection 🤔